src/StructuralSolvers/StructuralSolvers.jl

```diff
@@ -188,6 +192,31 @@ function tangent_matrix(st::AbstractStructuralState, alg::AbstractSolver) end
 # Solve function
 # ===============
 
+"""
+Solve a structural analysis problem with the given solver, returning a solution structure
+which holds the result and the algorithm used to obtain it.
+
+This function mutates the state defined in the analysis problem.
+Use [`solve`](@ref) to avoid mutation.
+For linear analysis problems, the algorithm doesn't need to be provided.
+"""
+function solve!(problem::AbstractStructuralAnalysis, solver::AbstractSolver=DummySolver())
+    _solve!(problem, solver)
+end
+solve!(pair::Tuple{AbstractStructuralAnalysis,AbstractSolver}) = solve!(pair.first, pair.last)
+
+"""
```

Comment:
Detail regarding style guide defined in CONTRIBUTING.md
```suggestion
""" Return initialized analysis """
```